src/main/java/com/es/phoneshop/DAO/CustomProductDao.java

```diff
@@ -46,20 +53,59 @@ public Optional<Product> getProductById(Long id) {
     }
 
     @Override
-    public List<Product> findProducts() {
+    public List<Product> findProducts(SortField sortField, SortOrder sortOrder, String query) {
```

Comment:
let's split the current method implementation for 2 methods: when the sorting/search are needed and when they are not needed
and depending on the fact if sort/search are needed implementation will look better